local.tf

```diff
@@ -10,31 +10,39 @@ locals {
   kubeconfig_name          = "${var.kubeconfig_name == "" ? "eks_${var.cluster_name}" : var.kubeconfig_name}"
 
   workers_group_defaults_defaults = {
```

Comment:
This is a little confusing as now we have the defaults for both worker group types mixed together.

How would someone reading this now what parameter is valid for each type of work group?

How about we leave `workers_group_defaults_defaults` as it is and create `workers_group_launch_template_defaults_defaults`? Does that make sense and work?

Reply:
I thought about that but it would result in some duplication if we wanna make it very separate if you're fine with that.
The other approach is to just merge the two defaults for the launch_template but I think that's as confusing as mixing the two.

Reply:
Hmmm I see your point about duplication.

At this point I really think the separation is important since launch template is a such a new feature and I don't want it to interfere with the current ASG/LC arrangement until everyone is confident in the ASG/LT approach.

So how about we have `workers_group_defaults_defaults` and `workers_group_launch_template_defaults_defaults` but the latter only contains the special values for LTs? e.g. `spot_max_price`, `spot_instance_pools` etc.

I think this will be clear and also won't change the existing code at all?

Reply:
I had a chance to think about this for a bit. I ended up just duplicating the defaults along.